### ISSUE TITLE
ApiGatewayV2 log group fails when changing api route path

### ISSUE THREAD

**Issue by LucasSGomide** [anomalyco/sst#5143]
Hey there,

First of all, I want to thank SST team for the great dev experience your product delivers!

I've been facing this issue for a while now and I'm not sure why. I took some time to go through some open and closed issues but still wasn't able to figure out what's going on.

### SST Versions:
- I originally faced the issue running `3.1.42` 
- Updated to `3.3.30` and got the same result

---

This is the error log I'm currently receiving:
```bash
ClientApiRouteTmcfhx sst:aws:ApiGatewayV2LambdaRoute → ClientApiRouteTmcfhxHandlerLogGroup aws:cloudwatch:LogGroup

sdk-v2/provider2.go:509: sdk.helper_schema: creating CloudWatch Logs Log Group (/aws/lambda/sustell-lucas-client-TestSST): operation error CloudWatch Logs: CreateLogGroup, https response error StatusCode: 400, RequestID: 0757aa15-8128-42d3-acfab4171e7c4916, ResourceAlreadyExistsException: The specified log group already exists: provider=aws@6.53.0
```

### Steps to reproduce the issue:

1. Create an `ApiGatetewayV2` resource
```typescript
const api = new sst.aws.ApiGatewayV2("ClientApi", {});
```

2. Declare routes
```typescript
api.route(
  "GET /resources/example/something",
  {
    handler:
      "./packages/functions/src/example.sstTestHandler",
    name: `${$app.name}-${$app.stage}-client-TestSST`,
  }
);
```
3. Run `sst dev`
4. Kill the development session
5. Update the API Route (from "/example" to "/example**s**")
```typescript
api.route(
  "GET /resources/examples/something",
  {
    handler:
      "./packages/functions/src/example.sstTestHandler",
    name: `${$app.name}-${$app.stage}-client-TestSST`,
  }
);
```
6. Run `sst dev`

---

### **Edit 1: Additional information**:

I believe that the issue is caused by defining a fixed name for the lambda function. When a change in the route path happens a new lambda function has to be deployed, and since I'm using the same name for that function, the log groups are already in place.

If SST state is able to control the deployment of the new lambda function after a path change, then I think it should be possible to control the log group behavior as well, it should either not be touched or be recreated.

Maybe this is actually an improvement request? Or is there already a solution to avoid this situation in current SST version?

---

I appreciate your help!

**Comment by jayair** [comment]
Yeah it seems like it's happening because you are setting the name explicitly. In this case the log group name derives from the function name. Partly why we discourage setting the name.

Workarounds would be comment out the route, do a deploy, add the new route and do a deploy. Or change the name of the function when you change the route.

I'll need to check with the team if this is something we can handle automatically.

**Comment by LucasSGomide** [comment]
@jayair thank you for the reply.

Is there any news regarding a fix/improvement for this issue?

**Comment by jayair** [comment]
I would do the workaround above. We'll need to look into if this is something that can be avoided in the long run.

**Comment by eikster-dk** [comment]
we are doing something like this in our sst.config.ts file
```
  async run() {
   .........

    $transform(sst.aws.Function, (args, opts) => {
      const handlerFileName = args.handler.split("/").pop().replace(".", "-");

      args.name ??= `${$app.stage}-${handlerFileName}`;

      args.logging ??= {
        logGroup: `${$app.stage}-${$app.name}`,
        format: "json",
      };
    });
  },
```

and we are hitting the same barrier as above where sst hangs for a while until til fails with the message that the function already exists. Any ideas of a better workaround on how to solve this @jayair ? I would really love to keep the simple naming convension, maybe hashing something behind the name that is related to the routeKey or whatever parent node that has created the function, but until now my attempts has failed me.

**Comment by jayair** [comment]
Yeah in this case you are taking over the naming responsibility from SST. So adding a hash should help here. But at that point, just let SST do it?
